python/test/test_api.py

```
@@ -0,0 +1,238 @@
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import asyncio
+import queue
+import time
+import unittest
+
+import numpy
+import pytest
+import tritonserver
+
+try:
+    import cupy
+except Exception:
+    cupy = None
+
+
+class TrtionServerAPITest(unittest.TestCase):
+    def test_not_started(self):
+        server = tritonserver.Server()
+        with self.assertRaises(tritonserver.InvalidArgumentError):
+            server.is_ready()
+
+    @pytest.mark.skipif(cupy is None, reason="Skipping gpu memory, cpupy not installed")
+    def test_gpu_memory(self):
+        import cupy
+
+        server = tritonserver.Server(
+            model_repository="/workspace/models", exit_timeout=5
+        )
+
+        server.start(blocking=True)
+
+        test = server.get_model("test")
+        fp16_input = cupy.array([[5], [6], [7], [8]], dtype=numpy.float16)
+        responses = test.infer(inputs={"fp16_input": fp16_input}, request_id="1")
+
+        for response in responses:
+            print(response)
+
+        responses = server.models["test"].infer(
+            inputs={"fp16_input": fp16_input}, request_id="1"
+        )
+
+        for response in responses:
+            print(response)
+        try:
+            pass
+        #            server.stop()
+        except Exception as error:
+            print(error)
+
+    def test_unload(self):
+        server = tritonserver.Server(
+            model_repository="/workspace/models",
+            exit_timeout=5,
+            model_control_mode=tritonserver.ModelControlMode.EXPLICIT,
+            startup_models=["test"],
+            log_verbose=True,
+            log_error=True,
+            log_info=True,
+        )
+        server.start(blocking=True)
+
+        model = server.models["test"]
+
+        responses = model.infer(
+            inputs={"fp16_input": numpy.array([[0.5]], dtype=numpy.float16)}
+        )
+
+        print(list(responses)[0])
```

Add assert to all these calls to fail instead of printing here and below?